pool/chainstate.go

```diff
@@ -398,6 +386,18 @@ func (cs *ChainState) handleChainUpdates(ctx context.Context) {
 					cs.cfg.Cancel()
 					continue
 				}
+
+				// Process mature payments without blocking.
+				go func(cfg *ChainStateConfig, height uint32, treasuryActive bool) {
```

Comment:
Why is this moved down? I don't think this is going to run unless the most recently mined block is one mined by the pool? (As we discussed previously on Matrix)

Reply:
Good catch, noticed the current call wasn't only running for pool mining mode, that's why it was moved. Rectifying.